src/ft2font.cpp

```diff
@@ -188,325 +189,126 @@ static FT_UInt ft_get_char_index_or_warn(FT_Face face, FT_ULong charcode)
+{
+    ft_outline_decomposer* d = reinterpret_cast<ft_outline_decomposer*>(user);
+    if (d->codes) {
+        *(d->vertices++) = control->x / 64.;
+        *(d->vertices++) = control->y / 64.;
+        *(d->vertices++) = to->x / 64.;
+        *(d->vertices++) = to->y / 64.;
+        *(d->codes++) = CURVE3;
+        *(d->codes++) = CURVE3;
+    }
+    d->index += 2;
+    return 0;
+}
 
-            case FT_CURVE_TAG_CONIC: // consume conic arcs
-            {
-                v_control.x = point->x;
-                v_control.y = point->y;
-
-            Do_Conic:
-                if (point < limit) {
-                    FT_Vector vec;
-                    FT_Vector v_middle;
-
-                    point++;
-                    tags++;
-                    tag = FT_CURVE_TAG(tags[0]);
-
-                    vec.x = point->x;
-                    vec.y = point->y;
-
-                    if (tag == FT_CURVE_TAG_ON) {
-                        double xctl = conv(v_control.x);
-                        double yctl = flip_y ? -conv(v_control.y) : conv(v_control.y);
-                        double xto = conv(vec.x);
-                        double yto = flip_y ? -conv(vec.y) : conv(vec.y);
-                        *(outpoints++) = xctl;
-                        *(outpoints++) = yctl;
-                        *(outpoints++) = xto;
-                        *(outpoints++) = yto;
-                        *(outcodes++) = CURVE3;
-                        *(outcodes++) = CURVE3;
-                        continue;
-                    }
-
-                    v_middle.x = (v_control.x + vec.x) / 2;
-                    v_middle.y = (v_control.y + vec.y) / 2;
-
-                    double xctl = conv(v_control.x);
-                    double yctl = flip_y ? -conv(v_control.y) : conv(v_control.y);
-                    double xto = conv(v_middle.x);
-                    double yto = flip_y ? -conv(v_middle.y) : conv(v_middle.y);
-                    *(outpoints++) = xctl;
-                    *(outpoints++) = yctl;
-                    *(outpoints++) = xto;
-                    *(outpoints++) = yto;
-                    *(outcodes++) = CURVE3;
-                    *(outcodes++) = CURVE3;
-
-                    v_control = vec;
-                    goto Do_Conic;
-                }
-                double xctl = conv(v_control.x);
-                double yctl = flip_y ? -conv(v_control.y) : conv(v_control.y);
-                double xto = conv(v_start.x);
-                double yto = flip_y ? -conv(v_start.y) : conv(v_start.y);
-
-                *(outpoints++) = xctl;
-                *(outpoints++) = yctl;
-                *(outpoints++) = xto;
-                *(outpoints++) = yto;
-                *(outcodes++) = CURVE3;
-                *(outcodes++) = CURVE3;
-
-                goto Close;
-            }
+static int ft_outline_cubic_to(
+  FT_Vector const* control1, FT_Vector const* control2, FT_Vector const* to, void* user)
```

Comment:
```suggestion
static int
ft_outline_cubic_to(FT_Vector const* control1, FT_Vector const* control2, FT_Vector const* to, void* user)
```
(Might need to wrap at 80)